app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,33 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    # @post = @user.posts
+    @post = @user.posts
   end
 
   def show
     @post = Post.find(params[:id].to_i)
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params[:id])
+    respond_to do |format|
+      format.html do
+        if @post_new.save
+          redirect_to "/users/#{@post_new.user_id}/posts/", message: 'Post created successfully'
+        else
+          render :new, status: 'Error: Post not created'
+        end
+      end
```

Comment:
`@post_new` has no reference
- [ ] Kindly make sure `@post_new` is declared variable

Reply:
@AshakaE Thanks i will look into that now and make necessary changes